scripts/download_data.py

```diff
@@ -0,0 +1,290 @@
+        }
+        baseUrl = self.getBamBaseUrl()
+        samples = self.args.samples.split(',')
+        for sample in samples:
+            samplePath = '{}/alignment/'.format(sample)
+            study = studyMap[sample]
+            fileName = (
+                '{}.mapped.ILLUMINA.bwa.{}.'
+                'low_coverage.20120522.bam'.format(sample, study))
+            sampleUrl = os.path.join(baseUrl, samplePath, fileName)
+            utils.log("Downloading index for '{}'".format(sampleUrl))
+            remoteFile = pysam.AlignmentFile(sampleUrl)
+            header = remoteFile.header
+            utils.log("Writing '{}'".format(fileName))
+            localFile = pysam.AlignmentFile(
+                fileName, 'wb', header=header)
+            refsToDownload = args.references.split(',')
+            for reference in refsToDownload:
+                utils.log("reference {}".format(reference))
+                iterator = remoteFile.fetch(
+                    reference.encode('utf-8'),
+                    start=self.chromMinMax.getMinPos(reference),
+                    end=self.chromMinMax.getMaxPos(reference))
+                for index, record in enumerate(iterator):
+                    if index >= args.num_reads:
+                        break
+                    localFile.write(record)
+                utils.log("{} records written".format(index))
+            iterator = None
+            remoteFile.close()
+            localFile.close()
+            baiFileName = fileName + '.bai'
+            os.remove(baiFileName)
+            utils.log("Indexing '{}'".format(fileName))
+            pysam.index(fileName.encode('utf-8'))
+        escapeDir()
+
+
+class NcbiFileDownloader(AbstractFileDownloader):
+    """
+    Downloads files from NCBI
+    """
+    def __init__(self, args):
+        super(NcbiFileDownloader, self).__init__(args)
+
+    def getBaseUrl(self):
+        return 'ftp://ftp-trace.ncbi.nih.gov/1000genomes'
+
+
+class EbiFileDownloader(AbstractFileDownloader):
+    """
+    Downloads files from EBI
+    """
+    def __init__(self, args):
+        super(EbiFileDownloader, self).__init__(args)
+
+    def getBaseUrl(self):
+        return 'ftp://ftp.1000genomes.ebi.ac.uk/vol1'
+
+
+sources = {
+    "ncbi": NcbiFileDownloader,
+    "ebi": EbiFileDownloader,
+}
+
+
+def parseArgs():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--source", default="ncbi", choices=sources.keys(),
+        help="the source to download from")
+    parser.add_argument(
+        "--dir-name", default="ga4gh-downloaded-data",
+        help="the name of the directory that the data is downloaded to")
+    parser.add_argument(
+        "--samples", default='HG00096,HG00533,HG00534',
+        help="a comma-seperated list of samples to download")
+    parser.add_argument(
+        "--num-reads", default=1000,
+        help="the number of reads to download per reference")
+    parser.add_argument(
```

Comment:
We should use this for the VCFs we download too: these are the chromosomes we're interested in, so we download the VCF, get the reads, and download the reference (later). So, I would change this to ``--chromosomes``. If the X and Y are awkward because of different file naming, then just forget about them